invenio_users_resources/services/users/service.py

```diff
@@ -286,3 +289,43 @@ def can_impersonate(self, identity, id_):
             raise PermissionDeniedError()
         self.require_permission(identity, "impersonate", record=user)
         return user.model.model_obj
+
+    def add_role(self, identity, id_, role_name):
+        """Add role to user."""
+        user = UserAggregate.get_record(id_)
+        if user is None:
+            # return 403 even on empty resource due to security implications
+            raise PermissionDeniedError()
+        self.require_permission(identity, "manage", record=user)
+        user.add_role(role_name)
+        return True
+
+    def remove_role(self, identity, id_, role_name):
+        """Remove role from user."""
+        user = UserAggregate.get_record(id_)
+        if user is None:
+            # return 403 even on empty resource due to security implications
+            raise PermissionDeniedError()
+        self.require_permission(identity, "manage", record=user)
+        user.remove_role(role_name)
+        return True
+
+    def list_roles(self, identity, id_):
```

Comment:
When discussing this one, I think I mixed `/api/groups` with `/api/users/<id>/groups`. I'm sorry.
Since you already have the roles loaded after doing `get_roles`, probably it'd be better to load them as `GroupAggregate` objects (`.from_model`) and use the [`GroupSchema`](https://github.com/ubiquitypress/invenio-users-resources/blob/master/invenio_users_resources/services/schemas.py#L131) to return a list (`many-True`).

We can talk IRL about this one, now that I wrap my head around it.

Reply:
I was trying to keep the api consistent using the search endpoint.